pkg/cosign/rego/rego.go

```diff
@@ -93,27 +100,57 @@ func ValidateJSONWithModuleInput(jsonBody []byte, moduleInput string) error {
 
 	evalQuery, err := r.PrepareForEval(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	var input interface{}
 	dec := json.NewDecoder(bytes.NewBuffer(jsonBody))
 	dec.UseNumber()
 	if err := dec.Decode(&input); err != nil {
-		return err
+		return nil, err
 	}
 
 	rs, err := evalQuery.Eval(ctx, rego.EvalInput(input))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	for _, result := range rs {
-		isCompliant, ok := result.Bindings[CosignEvaluationRule].(bool)
-		if ok && isCompliant {
-			return nil
+		switch response := result.Bindings[CosignEvaluationRule].(type) {
+		case []interface{}:
+			return evaluateRegoEvalMapResult(query, response)
+		case bool:
+			if response {
+				return nil, nil
+			}
 		}
 	}
 
-	return fmt.Errorf("policy is not compliant for query '%s'", query)
+	return nil, fmt.Errorf("policy is not compliant for query '%s'", query)
+}
+
+func evaluateRegoEvalMapResult(query string, response []interface{}) (warnings error, errors error) {
+	var warnMsg error
+	errMsg := fmt.Errorf("policy is not compliant for query '%s'", query)
+	for _, r := range response {
+		rMap := r.(map[string]interface{})
+		mapBytes, err := json.Marshal(rMap)
+		if err != nil {
+			return nil, fmt.Errorf("policy is not compliant for query '%s' due to parsing errors: %w", query, err)
+		}
+		var resultObject CosignRuleResult
+		err = json.Unmarshal(mapBytes, &resultObject)
+		if err != nil {
+			return nil, fmt.Errorf("policy is not compliant for query '%s' due to parsing errors: %w", query, err)
+		}
+		fmt.Printf("resultObject %v\n", resultObject)
+
+		// Check if it is complaint
+		if resultObject.Result {
+			return fmt.Errorf(resultObject.Warnings), nil
+		}
+		warnMsg = fmt.Errorf(resultObject.Warnings)
```

Comment:
`errors.New` since this will do weird things if `Warnings` has a formatting directive.

Reply:
Thanks for the review.